tokenizers/src/normalizers/mod.rs

```diff
@@ -1,11 +1,16 @@
 pub mod bert;
+pub mod precompiled;
+// Separatae file because of long bytes.
+#[cfg(test)]
+mod precompiled_test;
```

Comment:
I am not sure I know how to get a valid binary string otherwise that still manages to get some tests off.

This one is `spm` `nmt_nfkc` (default) normalization rule, which still deviates from the standard (tilde, and some right to left inconsistencies with nfkc). That's a great way to test this as they are the edge cases that cause problems with the tests listed.

Maybe I could reduce the file size by having the binary be directy encoded and not be u8 ? We don't care about readability anyway.

Reply:
We can maybe have it in a separate file, that we download specifically for the tests. We would need to extract the tests as integration, and download from the `Makefile`, but I prefer this solution than having this huge source code file.

Reply:
I don't think unit tests should rely on external files as much as possible.

I agree having this huge thing is not great, but I think downloading it as external is even worse. It make tests depend on an internet connexion, it makes logic of tests hard to understand, it can fail tests because of flaky connection, making tests flaky. It also hinder the test readability and so on.

I tried having this binary string in raw format within the source code (which bring the source file to ~233k) but it makes the file really unreadable with superduper long string, that because it's invalid utf8 messes with editors. 

The other option which I think is viable, is to split the Precompiled struct into an external crate that can tolerate this large file within it's own tests and we just implement the Normalizer trait in `tokenizers`. It's another way of making that downloadable, without actually being downloaded by the other crate.
It does add a dependency though :(

Reply:
Building tests in release mode (for bench for instance) with this huge file does mean compilation is *especially* slow.

Reply:
I extracted it as a separate crate. So I think this is correct.